### ISSUE TITLE
Partial expand

### ISSUE THREAD

**Issue by andreypopp** [hannesg/uri_template#7]
It would be nice to have partial expand feature which works by binding supplied values to variables and produces another template:

```
URLTemplate.new("{a}/{b}").partial_expand(a: 1)
=> <URLTemplate @pattern="1/{b}">
```

**Comment by hannesg** [comment]
Yep, I agree. I left it out at first because I didn't see the immediate value, but now I some pointful usecases.

**Comment by hannesg** [comment]
Okay, I've implemented something in the 0.6.0 release. Please test and report if something doesn't work like you expect it.

**Comment by andreypopp** [comment]
This is cool, though I expect

```
URITemplate.new('{+path}').expand_partial(path: 'x').to_s == 'x{+path}'
```

and

```
URITemplate.new('{?x,y}').expand_partial(path: 'x').to_s == '?x=1{&y}'
```

and so on...

**Comment by hannesg** [comment]
Hm, how should this case be handled:

```
URITemplate.new('{?x,y}').expand_partial(y: 'z')
```

`{?x}&y=z` is not guaranteed to issue a leading questionmark.

**Comment by andreypopp** [comment]
Do you think components inside `{?...}` should be expanded in order they are defined? I didn't find anything about it in [RFC6570](http://tools.ietf.org/html/rfc6570). I think that:

```
URITemplate.new('{?x,y}').expand_partial(y: 'z') # => URITemplate('?y=z{&x}')
```

would be reasonable.

**Comment by hannesg** [comment]
I guess you're right. Just because the example code uses the order of the variables doesn't mean it's mandatory.

**Comment by hannesg** [comment]
Hi, I've written some more code and some more specs. I've also grant you push access in case you want to add some more specs or change the behavior.

**Comment by hannesg** [comment]
Hi @andreypopp 

Have you tested the changed behavior? Can you give me some feedback?

**Comment by plexus** [comment]
Hi, I notice the behavior of URITemplate has changed between 0.6.0 and 0.7.0

``` ruby
URITemplate.new('/foo/{x}/{y}').expand_partial(x: 123).to_s
# 0.6.0 => "/foo/123/{y}"
# 0.7.0 => "/foo/123{x}/{y}"
```

is there an API to get the old behaviour? I'm using URITemplate for [Yaks](https://github.com/plexus/yaks) and this is blocking me from upgrading.
